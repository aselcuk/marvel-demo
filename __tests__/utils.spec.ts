const { sortDescByModifiedDate } = require('src/utils/utils');

describe('sortDescByModifiedDate', () => {
    describe('when the character comics list is given', () => {

        const comicBookItems = [
            {
                id: 40628,
                title: 'Hulk (2008) #55',
                modified: '2012-06-19T11:48:47-0400'
            },
            {
                id: 40630,
                title: 'Hulk (2008) #54',
                modified: '2012-06-11T17:31:00-0400'
            },
            {
                id: 40632,
                title: 'Hulk (2008) #53',
                modified: '2012-06-12T15:18:39-0400'
            }
        ];

        const result = sortDescByModifiedDate(comicBookItems);

        it('should be sorted in descending order by date', () => {
            expect(result).toMatchSnapshot();
        });
    });
});