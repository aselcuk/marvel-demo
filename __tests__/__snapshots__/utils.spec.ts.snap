// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortDescByModifiedDate when the character comics list is given should be sorted in descending order by date 1`] = `
Array [
  Object {
    "id": 40628,
    "modified": "2012-06-19T11:48:47-0400",
    "title": "Hulk (2008) #55",
  },
  Object {
    "id": 40632,
    "modified": "2012-06-12T15:18:39-0400",
    "title": "Hulk (2008) #53",
  },
  Object {
    "id": 40630,
    "modified": "2012-06-11T17:31:00-0400",
    "title": "Hulk (2008) #54",
  },
]
`;
